codec/reflectcodec/type_codec.go

```diff
@@ -344,6 +388,12 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 				maxSliceLen,
 			)
 		}
+		if omitEmpty {
+			p.PackBool(value.IsNil())
+			if value.IsNil() {
+				return p.Err
+			}
```

Comment:
It seems adding the `omitempty` tag has 2 effects for slices:
- causes `nil` and empty slices to be serialized differently
- causes `omitempty` tag to apply to the elements of the slice

is the first behavior desirable? either way could we add a comment that documents this?

Reply:
From my point of view, both things are desirable, specially that elements inside a slice/array inherit the `omitempty` tag.

Reply:
But @StephenButtolph and @aaronbuchwald would know best. I can update the documentation if that behaviour is acceptable.

Reply:
Removed the first (undesirable imo) behavior.